### ISSUE TITLE
hledger-ui build breaks with recent versions of fsnotify

### ISSUE THREAD

**Issue by peti** [simonmichael/hledger#820]
I have the following problem compiling hledger-ui 1.9.1 in Nix. I *believe* the issue is caused by `fsnotify`, but I am not entirely sure. The relevant part of the build log is:

~~~
[...]
Dependency HUnit -any: using HUnit-1.6.0.0
Dependency ansi-terminal -any: using ansi-terminal-0.8.0.4
Dependency async -any: using async-2.2.1
Dependency base -any: using base-4.11.1.0
Dependency base-compat -any: using base-compat-0.10.1
Dependency brick >=0.12: using brick-0.37.1
Dependency cmdargs -any: using cmdargs-0.10.20
Dependency containers -any: using containers-0.5.11.0
Dependency data-default -any: using data-default-0.7.1.1
Dependency directory -any: using directory-1.3.1.5
Dependency filepath -any: using filepath-1.4.2
Dependency fsnotify -any: using fsnotify-0.3.0.1
Dependency hledger -any: using hledger-1.9.1
Dependency hledger-lib -any: using hledger-lib-1.9.1
Dependency megaparsec -any: using megaparsec-6.4.1
Dependency microlens -any: using microlens-0.4.8.3
Dependency microlens-platform -any: using microlens-platform-0.3.9.0
Dependency pretty-show -any: using pretty-show-1.6.16
Dependency process -any: using process-1.6.3.0
Dependency safe -any: using safe-0.3.17
Dependency split -any: using split-0.2.3.3
Dependency text -any: using text-1.2.3.0
Dependency text-zipper -any: using text-zipper-0.10.1
Dependency time -any: using time-1.8.0.2
Dependency transformers -any: using transformers-0.5.5.0
Dependency vector -any: using vector-0.12.0.1
Dependency vty >=5.5: using vty-5.20
[...]
Hledger/UI/Main.hs:215:13: error:
    • The constructor ‘Modified’ should have 3 arguments, but has been given 2
    • In the pattern: Modified f _
      In a case alternative: Modified f _ -> f `elem` files
      In the expression:
        case fev of
          Modified f _ -> f `elem` files
          _ -> False
    |
215 |             Modified f _ -> f `elem` files
    |             ^^^^^^^^^^^^

Hledger/UI/Main.hs:238:15: error:
    • The constructor ‘Added’ should have 3 arguments, but has been given 2
    • In the pattern: Added f _
      In an equation for ‘showFSNEvent’:
          showFSNEvent (Added f _) = "Added " ++ show f
    |
238 | showFSNEvent (Added    f _) = "Added "    ++ show f
    |               ^^^^^^^^^^^^
~~~

**Comment by peti** [comment]
https://hydra.nixos.org/build/76296327/nixlog/45 has the complete build log.

**Comment by peti** [comment]
/me doesn't understand this error. The `Modified` constructor has 3 arguments -- even in the latest `fsnotify` version. :confused: I'll dig further ...

**Comment by peti** [comment]
Ah, okay. Now I see ... 236101e31a1cd6d771e070840b36018e763d8699 fixed this in git, but the 1.9.1 doesn't have that change yet.

**Comment by peti** [comment]
Fixed in https://github.com/peti/nixpkgs/commit/a110661699e546b0cc0d347adf2194ab6303ccd8.
